modules/apps/redirect/redirect-service/src/main/java/com/liferay/redirect/service/impl/RedirectEntryServiceImpl.java

```diff
@@ -102,40 +103,46 @@ public RedirectEntry fetchRedirectEntry(long redirectEntryId)
 	}
 
 	@Override
-	public List<RedirectEntry> getRedirectEntries(
-			long groupId, int start, int end,
-			OrderByComparator<RedirectEntry> orderByComparator)
+	public List<RedirectEntry> getRedirectEntries(long groupId)
 		throws PortalException {
 
-		PermissionChecker permissionChecker = getPermissionChecker();
+		List<RedirectEntry> redirectEntries = ListUtil.copy(
+			redirectEntryPersistence.findByGroupId(groupId));
+
+		Iterator<RedirectEntry> iterator = redirectEntries.iterator();
 
-		if (!permissionChecker.hasPermission(
-				groupId, RedirectEntry.class.getName(),
-				RedirectEntry.class.getName(), ActionKeys.VIEW)) {
+		while (iterator.hasNext()) {
+			RedirectEntry redirectEntry = iterator.next();
 
-			throw new PrincipalException.MustHavePermission(
-				permissionChecker, RedirectEntry.class.getName(),
-				RedirectEntry.class.getName(), ActionKeys.VIEW);
+			if (!_redirectEntryModelResourcePermission.contains(
+					getPermissionChecker(), redirectEntry, ActionKeys.VIEW)) {
+
+				iterator.remove();
+			}
 		}
 
-		return redirectEntryLocalService.getRedirectEntries(
-			groupId, start, end, orderByComparator);
+		return redirectEntries;
 	}
 
 	@Override
-	public int getRedirectEntriesCount(long groupId) throws PortalException {
-		PermissionChecker permissionChecker = getPermissionChecker();
+	public List<RedirectEntry> getRedirectEntries(
+			long groupId, int start, int end,
+			OrderByComparator<RedirectEntry> orderByComparator)
+		throws PortalException {
 
-		if (!permissionChecker.hasPermission(
-				groupId, RedirectEntry.class.getName(),
-				RedirectEntry.class.getName(), ActionKeys.VIEW)) {
+		List<RedirectEntry> redirectEntries = getRedirectEntries(groupId);
```

Comment:
@chileces @tototrinh Thank you for sending this change to us. Bascially, the Core-Infra team is in charge of the permission framework, not individual permission usages. So this PR should be more suitable to be reviewed by other teams. 

However, I do have a concern about this change, as it may introduce performance issue -- it fetches all entries related to the group every time, even if only a range is requested, which may be very heavy when the group has large number of redirect entries.

On the other hand, based on the discussion in the PTR ticket related to this LPS, I think the correct way is to make sure the Power User has VIEW permission for redirect entries of the certain site (group).